### DISCUSSION TITLE
https://angular.love/en/angular-18-whats-new

### DISCUSSION THREAD

**Discussion by giscus** [HouseOfAngular/angular-love#297] [General]
# https://angular.love/en/angular-18-whats-new

Angular.love - a place for all Angular enthusiasts created to inspire and educate.

https://angular.love/en/angular-18-whats-new

**Comment by bennadel** [comment]
Good write-up, thank you. I've historically been a fan of defining Inputs / Outputs in the `@Comonent()` meta-data. I always felt like using the `@Input` decorators a bit like mixing responsibilities. But, now that the inputs and outputs don't need decorators, I might try to use them. Do you find yourself grouping inputs / outputs together? For example, something like:

```
class MyComponent {
   // ... private variables for dependency injection ...

   // ... input variables ...
   // ... output variables ...

   // ... public variables for internal state + template rendering ...
}
```

Just curious if you have a strategy that has been feeling good? Or, maybe you're just mixing all the variables together?
